File: II sem/Programowanie Obiektowe/Projekty/projekt1/POproject1/WilczeJagody.cpp
#include "Swiat.h"
#include "WilczeJagody.h"

WilczeJagody::WilczeJagody(int x, int y) : Roslina(EMOJI ? u8"WJ" : "WJ", 99, x, y)
{

}

void WilczeJagody::kolizja(Organizm* inny)
{
	string nazwa = typeid(*inny).name();
	swiat->dodajLog(this, " Wilcze jagody zatruly " + nazwa.substr(6));
	zabijanie();
	inny->zabijanie();

}

Organizm* WilczeJagody::potomek()
{
	return new WilczeJagody(*this);
}

File: II sem/Programowanie Obiektowe/Projekty/projekt2/po3/src/WilczeJagody.java
public class WilczeJagody extends Roslina
{
    public WilczeJagody(String ikonka, int x_pos, int y_pos, int sila, int inicjatywa, Swiat swiat) {
        super(ikonka, x_pos, y_pos, sila, inicjatywa, swiat);
    }

    public WilczeJagody(Organizm organizm) {
        super(organizm);
    }

    @Override
    void kolizja(Organizm organizm) {

    }

    @Override
    public void moc(Organizm organizm) {
        organizm.zabijanie();
    }

    @Override
    Organizm kopiuj(Organizm organizm) {
        return new WilczeJagody(organizm);
    }
}
